packages/opds-web-client/src/components/DownloadButton.tsx

```diff
@@ -107,14 +108,7 @@ export default class DownloadButton extends React.Component<
   }
 
   fileExtension() {
-    return (
-      {
-        "application/epub+zip": ".epub",
-        "application/pdf": ".pdf",
-        "application/vnd.adobe.adept+xml": ".acsm",
-        "application/x-mobipocket-ebook": ".mobi"
-      }[this.mimeType()] || ""
-    );
+    return typeMap[this.mimeType()]?.extension ?? "";
```

Comment:
So, basically, if the left side is ever null or undefined, then and only then return `""`, right?

Reply:
For Typescript syntax that is new and/or is not already in the repo, do you mind writing a small comment about it? It'll make it easier to read.

Reply:
Will do, and yes that's right. It's very similar to doing ` const blah = maybeDefined || fallback`, except that it only uses the fallback for `null` and `undefined`, while ` || ` will also fall back for falsy values (0, false, etc).